sync/src/synchronizer/block_fetcher.rs

```diff
@@ -151,9 +141,22 @@ impl BlockFetcher {
             let mut header = self
                 .active_chain
                 .get_ancestor(&best_known_header.hash(), start + span - 1)?;
+
+            // Judge whether we should fetch the target block, neither stored nor in-flighted
             for _ in 0..span {
                 let parent_hash = header.parent_hash();
-                if should_fetch(header.hash()) {
+                // NOTE: Filtering `BLOCK_STORED` but not `BLOCK_RECEIVED`, is for avoiding
+                // stopping synchronization even when orphan_pool maintains dirty items by bugs.
+                let stored = self
+                    .active_chain
+                    .contains_block_status(&parent_hash, BlockStatus::BLOCK_STORED);
+                if stored {
```

Comment:
```
if stored {
  break;
} else if inflight.insert(...) {
  fetch.push(header);  
}
```

or 

```
if stored {
  break;
}
if inflight.insert(...) {
  fetch.push(header);  
}
```

Reply:
I personally prefer giving the `insert` result a variable name. I never know whether the `true` returns from `insert` means the key already exists or the key has successfully inserted.  

🤣

Reply:
> I personally prefer giving the `insert` result a variable name. I never know whether the `true` returns from `insert` means the key already exists or the key has successfully inserted.
> 

may refactoring on download scheduler